### DISCUSSION TITLE
PyRight is expanding getattr default argument type to the attribute type

### DISCUSSION THREAD

**Discussion by Andrej730** [microsoft/pyright#9401] [Q&A]
It seems PyRight is deducing that a.y is probably have similar type to the fallback value which possibly gets in the way. Is it considered a bug?

```python
if x := getattr(a, "y", False):
    reveal_type(x) # Type of "x" is "Any | Literal[True]"
```

**Comment by erictraut** [comment]
Pyright is working correctly here. The expression `getattr(a, "y", False)` evaluates to type `Any | bool` based on the overload provided in the typeshed stubs. This value is assigned to `x`, so it also has the type `Any | bool`. The type of `x` is then narrowed based on a truthy type guard check, which means that its type is narrowed to `Any | Literal[True]`.
